src/pages/index.js

```diff
@@ -0,0 +1,333 @@
+const cardSubmitButton = cardModal.querySelector(".modal__submit-button");
+const cardModalLinkInput = cardModal.querySelector("#add-card-link-input");
+const cardModalCaptionInput = cardModal.querySelector("#add-card-name-input");
+
+// Avatar form element
+const avatarModal = document.querySelector("#edit-avatar-modal");
+const avatarFormElement = avatarModal.querySelector(".modal__form");
+const avatarModalCloseButton = avatarModal.querySelector(
+  ".modal__close-button"
+);
+const avatarSubmitButton = avatarModal.querySelector(".modal__submit-button");
+const avatarInput = avatarModal.querySelector("#profile-avatar-input");
+
+// card related elements
+const cardTemplate = document.querySelector("#card-template");
+const cardsList = document.querySelector(".cards__list");
+
+// Delete elements
+const deleteModal = document.querySelector("#delete-modal");
+const deleteFormElement = document.querySelector("#delete-form");
+
+// Preview image popup elements
+const previewModal = document.querySelector("#preview-modal");
+const previewModalImageEl = previewModal.querySelector(".modal__image");
+const previewModalCaptionEl = previewModal.querySelector(".modal__caption");
+const previewModalCloseButton = previewModal.querySelector(
+  ".modal__close-button-preview"
+);
+
+const closeButtons = document.querySelectorAll(".modal__close");
+
+function handleLike(evt, id, cardLikeButton) {
+  const isLiked = cardLikeButton.classList.contains("card__like-button_liked");
+
+  api
+    .changeLikeStatus(id, isLiked)
+    .then(() => {
+      cardLikeButton.classList.toggle("card__like-button_liked");
+    })
+    .catch(console.error);
+}
+
+let selectedCard, selectedCardId;
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", modalEscClose);
+  modal.addEventListener("mousedown", modalOverlayClose);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", modalEscClose);
+  modal.removeEventListener("mousedown", modalOverlayClose);
+}
+
+function modalEscClose(evt) {
+  if (evt.key === "Escape") {
+    const modal = document.querySelector(".modal_opened");
+    closeModal(modal);
+  }
+}
+
+function modalOverlayClose(evt) {
+  if (evt.target.classList.contains("modal")) {
+    closeModal(evt.target);
+  }
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+
+  const submitButton = evt.submitter;
+
+  setButtonText(submitButton, true, "Save", "Saving...");
+
+  const updatedUserInfo = {
+    name: editModalNameInput.value,
+    about: editModalDescriptionInput.value,
+  };
+
+  api
+    .editUserInfo(updatedUserInfo)
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitButton.textContent = "Save";
```

Comment:
`      submitButton.textContent = "Save";
`

Please, reuse the `setButtonText` function, to avoid code duplication

Please, don't forget to use the right button in each  case. You have `submitButton` for the profile submit button in line 161

```js

    .finally(() => {
      setButtonText(submitButton, false);
    });
```